test/test_slack_model.jl

```diff
@@ -47,12 +47,14 @@
     # slack variables do not influence objective value
     @test isapprox(obj(model, x[1:n]), obj(smodel, x), rtol = rtol)
     @test neval_obj(model) == 2
+    @test neval_obj(smodel) == 2
```

Comment:
Because of [this](https://github.com/JuliaSmoothOptimizers/NLPModelsModifiers.jl/blob/master/src/slack-model.jl#L146), defined [here](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/master/src/nlp/utils.jl#L118), `neval_obj(slack)` is defined as `neval_obj(slack.model)`. The same goes for `increment!`.
That means that this test is exactly the same as the line above.

What should be tested is that `slack.counters === slack.model.counters`.